org.mixedrealitytoolkit.uxcore/Experimental/NonNativeKeyboard/NonNativeKeyTouchAdapter.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+using System.Collections;
+using MixedReality.Toolkit;
+using MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.UX.Experimental
+{
+    public class NonNativeKeyTouchAdapter : MonoBehaviour
+    {
+        private const float ColliderMargin = 30.0f;
+        private const float ColliderThickness = 70.0f;
+        private const float ColliderZDelta = 20.0f;
+        private const float ReClickDelayTime = 1.0f;
+        private const float AnimationTime = 0.2f;
+        private const float AnimationMovementDelta = 20.0f;
+
+        private StatefulInteractable interactable;
+        private Graphic image;
+        private float lastClickTime;
+        private bool isInitialized;
+        private Vector3 defaultPosition;
+        private Vector3 animatedPosition;
+        private BoxCollider buttonCollider;
+        private Vector3 buttonColliderDefaultCenter;
+        private Color defaultImageColor;
+
+        private void Awake()
+        {
+            defaultPosition = transform.localPosition;
+            animatedPosition = defaultPosition + new Vector3(0, 0, AnimationMovementDelta);
+        }
+
+        private void OnEnable()
+        {
+            transform.localPosition = defaultPosition;
+            lastClickTime = Time.time;
+            if (isInitialized)
+            {
+                 buttonCollider.center = buttonColliderDefaultCenter;
+            }
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            if (isInitialized)
+            {
+                return;
+            }
+            isInitialized = true;
+
+            var rectTransform = GetComponent<RectTransform>();
+            buttonCollider = gameObject.EnsureComponent<BoxCollider>();
```

Comment:
I don't think we should be automatically creating a box collider here.

Reply:
Uhm... I think we do. Otherwise, how will it be touchable otherwise?

Reply:
let's just call "get component"  I would like the prefab to explicitly define their own collider.   Let's follow the pattern used by XRI.   Perfab can specify colliders via serialized field, if not specific call get components in children

Example from XRBaseInteractable
```cs

        /// <summary>
        /// See <see cref="MonoBehaviour"/>.
        /// </summary>
        protected virtual void Awake()
        {
            // If no colliders were set, populate with children colliders
            if (m_Colliders.Count == 0)
            {
                GetComponentsInChildren(m_Colliders);
                // Skip any that are trigger colliders since these are usually associated with snap volumes.
                // If a user wants to use a trigger collider, they must serialize the reference manually.
                m_Colliders.RemoveAll(col => col.isTrigger);
            }
```

Reply:
I will repeat: I can't change anything to the _prefab of a key_ because _a key is not a prefab_. All keys are just gameobjects in the keyboard. See? No blue cubes
<img width="199" alt="image" src="https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/4129183/faf11a1d-f40f-4c7f-bb38-6ac95b0ffb84">

So I have to add every piece of infrastructure myself. And if I, per your suggestion, do not create the collider, the result it the touch no longer works. The unit test fails. 

<img width="296" alt="image" src="https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/4129183/9e768b3a-b02d-45da-b09d-c44be047cfbd">


Please try it yourself. I have uploaded, for your convenience, a version of NonNativeKeyTouchAdapter.cs that has all the collider stuff deleted. 
[NonNativeKeyTouchAdapter.zip](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/files/14513655/NonNativeKeyTouchAdapter.zip)